src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,18 @@
+package duke;
+
+/**
+ * An abstract class representing a parser.
+ * 
+ * @param <T> The type of Object for the string to be parsed into.
+ */
+public abstract class Parser<T> {
```

Comment:
This is elegant, but shall we use `interface` instead of `abstract class` here as you are only defining behaviours here?

Reply:
May I ask for the reason? In my opinion, in the case I want to extend it further to include some variables, which might be the case, it seems an abstract class would work better, but do let me know if you disagree.

Reply:
hem its fine if you want to extend it in the future, I think its just my preference to not add something that I wont be certain I will add it or not (like just focus on the present first), but it really depends on the scope of the project